Clarify `error` status (#462)
exercism/docs: building/tooling/test-runners/interface.md

- `pass`: All tests passed
- `fail`: At least one test has the status `fail` or `error`

> - `error`: No test was executed (this usually means e.g. a compile error, or a syntax error)

Reviewer: ```suggestion
- `error`: No test was executed (this usually means a compile error or a syntax error)
```

(It's not very usual to use `e.g.` in the middle of a sentence)

> To clarify, `error` should _only_ be used if **none of the tests were run**.

Reviewer: ```suggestion
The `error` status should _only_ be used if **none of the tests were run**.
```

> Usually, this only happens for compiled languages, as not being able to compile the code will prevent the tests from being run.

Reviewer: ```suggestion
For compiled languages this is generally a result of the as code not being able to compile.
```

Author: ```suggestion
For compiled languages this is generally a result of the code not being able to compile.
```

> In most interpreted languages, the tests will run even with syntax errors in the source, as they'll only find that out at run time.

Reviewer: ```suggestion
For interpreted languages, this is generally the result of a syntax error that stops the file from parsing.
```
